app/Platform/Services/TriggerDecoderService.php

```diff
@@ -0,0 +1,552 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Platform\Services;
+
+use App\Models\MemoryNote;
+use Illuminate\Support\Str;
+
+class TriggerDecoderService
+{
+    private function parseOperand(string $mem): array
+    {
+        $end = strlen($mem);
+        if ($end === 0) {
+            return ['Invalid', '', '', ''];
+        }
+
+        $type = '';
+        switch ($mem[0]) {
+            case 'd': case 'D': $type = 'Delta'; $mem = substr($mem, 1); $end--; break;
+            case 'p': case 'P': $type = 'Prior'; $mem = substr($mem, 1); $end--; break;
+            case 'b': case 'B': $type = 'BCD'; $mem = substr($mem, 1); $end--; break;
+            case '~':           $type = 'Inverted'; $mem = substr($mem, 1); $end--; break;
+        }
+
+        $size = '';
+        if ($end > 3 && $mem[0] === '0' && $mem[1] === 'x') {
+            switch ($mem[2]) {
+                case 'h': case 'H': $size = '8-bit'; break;
+                case ' ':           $size = '16-bit'; break;
+                case 'x': case 'X': $size = '32-bit'; break;
+
+                case 'm': case 'M': $size = 'Bit0'; break;
+                case 'n': case 'N': $size = 'Bit1'; break;
+                case 'o': case 'O': $size = 'Bit2'; break;
+                case 'p': case 'P': $size = 'Bit3'; break;
+                case 'q': case 'Q': $size = 'Bit4'; break;
+                case 'r': case 'R': $size = 'Bit5'; break;
+                case 's': case 'S': $size = 'Bit6'; break;
+                case 't': case 'T': $size = 'Bit7'; break;
+                case 'l': case 'L': $size = 'Lower4'; break;
+                case 'u': case 'U': $size = 'Upper4'; break;
+                case 'k': case 'K': $size = 'BitCount'; break;
+                case 'w': case 'W': $size = '24-bit'; break;
+                case 'g': case 'G': $size = '32-bit BE'; break;
+                case 'i': case 'I': $size = '16-bit BE'; break;
+                case 'j': case 'J': $size = '24-bit BE'; break;
+
+                case '0': case '1': case '2': case '3': case '4':
+                case '5': case '6': case '7': case '8': case '9':
+                case 'a': case 'b': case 'c': case 'd': case 'e': case 'f':
+                case 'A': case 'B': case 'C': case 'D': case 'E': case 'F':
+                    // no size specified, implied 16-bit. convert to explicit
+                    $size = '16-bit';
+                    $mem = substr($mem, 0, 2) . ' ' . substr($mem, 2);
+                    break;
+
+                default:
+                    $size = $mem[2];
+                    break;
+            }
+
+            $mem = substr($mem, 3);
+        } elseif ($end > 2 && $mem[0] === 'f' || $mem[0] === 'F') {
```

Comment:
There may be a subtle bug here. This is evaluated as:

```php
// ((A && B) || C)
(($end > 1 && $mem[0] === 'f') || $mem[0] === 'F') {
```

However, I believe the intent is for this to be evaluated as `(A && (B || C))`.

This also occurs with the subsequent `elseif` comparing `$mem[0]` to "h" or "H".